System.Compiler/Reader.cs

```diff
@@ -1924,6 +1924,10 @@ private AssemblyNode.PostAssemblyLoadProcessor ReferringAssemblyPostLoad
     private TypeNodeList GetTypeParametersFor(int parentIndex, Member parent){
       GenericParamRow[] genericParameters = this.tables.GenericParamTable;
       TypeNodeList types = new TypeNodeList();
+      TypeNodeList savedTypes = this.currentTypeParameters;
+      this.currentTypeParameters = types;
+      try
+      {
       int i = 0, n = genericParameters.Length, j = n-1;

```

Comment:
I have some doubts about using that (especially considering I'm working on an alternative that doesn't require Visual Studio), but either way it would be a separate issue to consider from this pull request.